### PR TITLE
#160451597 reset password via email

### PR DESCRIPTION
**Pull request by Victor-Ugwueze**
#### What does this PR do?
- Reset password via email

#### Description of Task to be completed?
  - add AuthController.js
  - add ```resetPasseord``` and ```forgotPassword``` methods to AuthController
  - add authValidator.js to middleware to verify token from user's request and check the new password validity.
 - add route `/api/auth/forgot_password` and `/api/auth/reset_password`
  

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160451597](https://www.pivotaltracker.com/story/show/160451597 "reset password")
 
### Question
 NA

### CODE REVIEW

File: server/controllers/AuthController.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        // Send mail
+        EmailController.validationEmail(result);
+        return response.status(200)
+          .json({
+            status: 'success',
+            message: 'Password reset details has been sent to your mail',
+            link,
+          });
+      })
+      .catch(e => e);
```
Comment (KaiserPhemi) [inline server/controllers/AuthController.js]: #### Critical
What's `e`? Be more explicit. 
`errors` or `error` is clearer

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
#### Critical
- Implement feedback
- Fix merge conflict

**Comment by Victor-Ugwueze** [comment]
I have fixed the conflict. I also changed package manager to yarn,  but I am having import module error.




screen shot 2018-10-09 at 1 07 30 am

**Comment by Victor-Ugwueze** [comment]
I have Implemented the feedback. Please review @KaiserPhemi, thank you.

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**